src/App.tsx

```diff
@@ -17,18 +67,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedFilter={selectedFilter}
+                setSelectedFilter={setSelectedFilter}
+                setSearchTerm={setSearchTerm}
+                searchTerm={searchTerm}
+              />
             </div>
-
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0 ? (
+                <TodoList
+                  todos={filteredTodo}
+                  setSelectedTodo={setSelectedTodo}
+                  selectedTodo={selectedTodo}
+                />
+              ) : (
+                <Loader />
+              )}
```

Comment:
The logic was implemented, sir!

Reply:
> The logic was implemented, sir!

In your logic nothing changes, you remove the loader only in case you have at least one todo in array. I've added the comment in `useEffect`